Java/core/src/main/java/com/amazon/randomcutforest/sampler/CompactSampler.java

```diff
@@ -418,104 +290,60 @@ private void swapWeights(int a, int b) {
         }
     }
 
-    public static class Builder<T extends CompactSampler.Builder<T>> {
+    public static class Builder extends AbstractStreamSampler.Builder<Builder> {
```

Comment:
I think this change means that we won't be able to further subclass CompactSampler.Builder and have it work as expected.

Reply:
Done.